pkg/descheduler/strategies/duplicates.go

```diff
@@ -67,11 +69,30 @@ func RemoveDuplicatePods(
 		// If any of the existing lists for that first key matches the current pod's list, the current pod is a duplicate.
 		// If not, then we add this pod's list to the list of lists for that key.
 		duplicateKeysMap := map[string][][]string{}
+	PodLoop:
 		for _, pod := range pods {
 			ownerRefList := podutil.OwnerRef(pod)
 			if hasExcludedOwnerRefKind(ownerRefList, strategy) {
 				continue
 			}
+			for _, ownerRef := range pod.OwnerReferences {
```

Comment:
Could this just be done in the ownerRef loop that's already going through them below this? (L97)

Reply:
I have extracted that into a separate method for better readability.

Reply:
My point is that the changes in https://github.com/kubernetes-sigs/descheduler/pull/334/files#diff-f2a21c2960284e0b92eea0f1afd7ea40R125-R138 could be done in the ownerref loop that is already there (https://github.com/kubernetes-sigs/descheduler/pull/334/files/5e089eeeededd3ccfc2f389d9c14e5be9eeb6a5d#diff-f2a21c2960284e0b92eea0f1afd7ea40R97), otherwise you're looping through the ownerrefs twice in a row. Breaking it into a separate function doesn't address that

Reply:
@damemi Fixed this